src/chromium/cargo.md

```diff
@@ -1,56 +1,53 @@
-# Using cargo for experimental tools
+# Comparing Chromium and Cargo ecosystems
 
-Subjectively,
+Rust community typically uses `cargo` and libraries from [crates.io][2].
+Chromium is built using `gn` and `ninja` and a curated set of dependencies.
 
-```bob
-High        ^
-            |      x cargo
-            |
-Development |                     x "cargo --offline"
-speed       |
-            |                                  x  "gn/ninja"
-            |                                     "rust_executable(...)" 
-Low         +---------------------------------------------------->
-             Low               Determinism                 High
-```
-
-`cargo` works great for pure-Rust tools, but isn't optimized for large multi-
-language projects like Chromium. Chromium uses `gn` and `ninja`.
-
-When writing a tool in Rust, your choices are:
+When writing code in Rust, your choices are:
 
 * Use `gn` and `ninja` (using the `rust_executable` template we'll meet
   later)
 * Use `cargo`, but [restrict yourself to Chromium's audited toolchain and crates][0]
 * Use `cargo`, trusting a [toolchain][1] and [crates downloaded from the internet][2]
 
-Your organization's policy, and/or common sense, may prohibit you from doing
-these things.
-
 From here on we'll be focusing on `gn` and `ninja`.
 
 ## Mini exercise
 
-Discuss in small groups the policies within your own team and organization,
-and come to a group agreement about what's an acceptable level of risk.
+Discuss in small groups the policies within your own team and
+organization, and come to a group agreement about your assessment
+of risk associated with the choices above.
 
 <details>
 
-Talk about the cultural differences between the `cargo` world and the Chromium
-world: for instance, that code reuse is cheap in `cargo` and that it's
-encouraged to use (and create) lots of small single-purpose crates, which is
-difficult in Chromium (for good reasons). The net effect is that `cargo`
-ecosystem development feels more agile.
-
-Explain that it might seem strange to write tools in Rust, but this is
-increasingly popular across the industry --- Rust tools are quicker and work
-more reliably.
-
 Assuming folks taking the course are physically together, ask them to discuss
 in small groups of 3-4 people. Then, ask each table whether they've come
 to a consensus on the level of risk.
 
 Later in the course, we'll be running an actual `cargo`-based tool, `gnrt`.
+(`gnrt` uses Chromium's copy of `cargo` and `rustc`, but depends on third-party
+libraries downloaded from the internet, although `run_gnrt.py` asks `cargo` that
+only `--locked` contents are allowed via `Cargo.lock`.)
+
+Depending on the discussion, consider also mentioning the points below:
+
+* Talk about the cultural differences between the `cargo` world and the Chromium
+  world. Point out that one reason for adopting Rust in Chromium is to tap into
```

Comment:
I think we need to explicitly talk about how cargo development _feels_ relative to Chromium development. I especially don't like the "Point out" part. It's factually correct, but it seems a distraction — in a cargo world it's _trivial_ to pull in extra libraries and thus it's very quick to iterate on a project. In Chromium will still have a lot of intentional checks and balances, no matter how good our gnrt + gn tooling is. It's very slow to iterate relative to cargo.

I think this is one of the most important things to get across in the whole Chromium section of this course.